feat(extra-natives/five): support for cl_crosshair_t command

JohnDeved opened the pull request:
### Goal of this PR

This PR aims to implement support for the `cl_crosshair_t` [command from CS:GO](https://totalcsgo.com/commands/clcrosshairt). The `cl_crosshair_t` command allows players to customize their crosshair to a "T" style by removing the top line of the crosshair.

### How is this PR achieving the goal

- **Command Implementation**: Added the `cl_crosshair_t` command to toggle the "T" style crosshair.
  ```cpp
  static ConVar<bool> cl_crosshair_t("cl_crosshair_t", ConVar_Archive, false);
  ```

- **Drawing Logic Update**: Modified the `DoCrosshairDraw` function to conditionally skip drawing the top vertical line of the crosshair when `cl_crosshair_t` is enabled.
  ```cpp
  // draw top vertical crosshair line if cl_crosshair_t is disabled
  if (!cl_crosshair_t.GetValue())
  {
      int iInnerTop = iCenterY - iInnerCrossDist - iBarThickness / 2;
      int iOuterTop = iInnerTop - iBarSizeInner;
      DrawCrosshairRect(r, g, b, flLineAlphaInner, x0, iOuterTop, x1, iInnerTop, bAdditive);
  }
  ```

- **Whitelist Update**: Added `cl_crosshair_t` to the `g_prodCommandsWhitelist` to ensure the command is recognized and can be used.
  ```cpp
  static std::vector<std::string> g_prodCommandsWhitelist{
      ...
      "cl_crosshair_t" // Added command
  };
  ```

### This PR applies to the following area(s)

- FiveM

### Successfully tested on

**Game builds:** 

**Platforms:** Windows

### Checklist

- [ ] Code compiles and has been tested successfully.
- [x] Code explains itself well and/or is documented.
- [x] My commit message explains what the changes do and what they are for.
- [ ] No extra compilation warnings are added by these changes.

FabianTerhorst reviewed (approved):
Thanks for the contribution. Can you change your commit messages to match our commit standard:

```
feat(extra-natives/five): support for cl_crosshair_t command
```

FabianTerhorst commented:
Will manually change the commit message.

JohnDeved commented:
hey. thanks for the response, I left this PR on draft because I haven't come around to building and testing these changes.
I squashed the commits and changed the commit message

FabianTerhorst commented on code/components/extra-natives-five/src/DrawImNatives.cpp:
```diff
@@ -3498,3 +3504,4 @@ static void DoCrosshairDraw()
 
 	finishBatch();
 }
+
```
Please remove the extra space.

JohnDeved replied:
> Please remove the extra space.

done

FabianTerhorst reviewed (approved):
👍

JohnDeved commented:
@FabianTerhorst hey I just noticed this PR introduces a small drawing issue with the bottom line of the crosshair. I created a PR to fix this issue #2733
